### PR TITLE
Add `FORGET PARTITION` query to remove old partition nodes from ZooKeeper

### LINKED ISSUE #58925: Implement `SYSTEM` query to drop old partition nodes from ZooKeeper
In ReplicatedMergeTree we have a special nodes to allocate block numbers in each partition. These nodes are never removed since creation because it can lead to race condition with INSERT queries in the following way:
```
r1 (memory): 202210_0_777_99
r2 (memory): 202210_0_777_99
zookeeper: /blocks/202210 -> cversion = 778
Now someone start looong SELECT query from r1. 
Now someone run ALTER TABLE r2 DROP PARTITION 202210
r1 (memory): 202210_0_777_99 -- cannot remove part, because someone reading from it
r2 (memory): nothing -- last part was dropped
zookeeper: /blocks/_nothing_, we removed partition because last part was removed
Now someone INSERT new data part to r2
r1 (memory): 202210_0_777_99 -- cannot remove part, because someone reading from it
r2 (memory): 202210_0_0_0 new data part 
zookeeper: /blocks/202210 -> cversion = 1 freshly recreated node with new cversion
Now r1 pulls new state into memory and see
202210_0_0_0 intersects old part 202210_0_777_99 (someone still read from it)
```

The idea is to implement `SYSTEM` query which will cleanup old nodes from `blocks/partition`. When client will be sure that no new INSERT may appear in the old partition it's safe to execute this command. Syntax:
`SYSTEM DROP PARTITION NODE table partition` (suggestions are welcome).

### PR DESCRIPTION
**Pull request by serxa**
Fixes https://github.com/ClickHouse/ClickHouse/issues/58925

### Changelog category (leave one):
- New Feature

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Added query `ALTER TABLE table FORGET PARTITION partition` that removes ZooKeeper nodes, related to an empty partition.

### Documentation entry for user-facing changes

- [x] Documentation is written (mandatory for new features)

### CODE REVIEW

**Review by Userlugard — APPROVED** [review]

**Comment by serxa** [comment]
Checked it manually. Works perfectly. Working on a test...

File: src/Storages/StorageReplicatedMergeTree.cpp [hunk 1]
```diff
@@ -7259,6 +7259,20 @@ void StorageReplicatedMergeTree::fetchPartition(
...
+{
+    assertNotReadonly();
+    zkutil::ZooKeeperPtr zookeeper = getZooKeeperAndAssertNotReadonly();
+
+    String partition_id = getPartitionIDFromQuery(partition, query_context);
+    LOG_INFO(log, "Forget partition {}", partition_id);
+
+    String block_numbers_path = fs::path(zookeeper_path) / "block_numbers";
+    String partition_path = fs::path(block_numbers_path) / partition_id;
+    zookeeper->remove(partition_path);
```
Comment (alesapin) [inline src/Storages/StorageReplicatedMergeTree.cpp:7272]: What if partition doesn't exists? Maybe catch error and provide more clear exception?

File: docs/en/sql-reference/statements/alter/partition.md [hunk 2]
```diff
@@ -9,6 +9,7 @@ The following operations with [partitions](/docs/en/engines/table-engines/merget
 
 - [DETACH PARTITION\|PART](#detach-partitionpart) — Moves a partition or part to the `detached` directory and forget it.
 - [DROP PARTITION\|PART](#drop-partitionpart) — Deletes a partition or part.
+- [FORGET PARTITION](#forget-partition) — Deletes a partition completely from ZooKeeper.
```
Comment (alesapin) [inline docs/en/sql-reference/statements/alter/partition.md:12]: I would prefer `Deletes a partition metadata from zookeeper if it's empty`.

File: src/Storages/StorageReplicatedMergeTree.cpp (hunk 1 above)
Comment (serxa) [reply src/Storages/StorageReplicatedMergeTree.cpp:7272]: Okay, Let make it also `CANNOT_FORGET_PARTITION` with explanation "Unknown partition"?

**Review by alesapin — APPROVED** [review]

**Comment by serxa** [comment]
AST fuzzer (ubsan) - https://github.com/ClickHouse/ClickHouse/pull/60138

**Comment by serxa** [comment]
AST fuzzer (msan) — https://github.com/ClickHouse/ClickHouse/issues/60352

**Comment by serxa** [comment]
test_backup_restore_s3/test.py::test_user_specific_auth - Flaky
test_reload_clusters_config - Flaky

**Pull request merged**